src/sentinel.c

```diff
@@ -3161,11 +3161,13 @@ void sentinelConfigSetCommand(client *c) {
         sentinel.announce_port = numval;
     } else if (!strcasecmp(o->ptr, "sentinel-user")) {
         sdsfree(sentinel.sentinel_auth_user);
-        sentinel.sentinel_auth_user = sdsnew(val->ptr);
+        sentinel.sentinel_auth_user = strcmp(val->ptr,"") ? 
+            sdsnew(val->ptr) : NULL;
```

Comment:
how about using `sdslen() == 0` instead of `strcmp`? and also maybe `sdsdup` instead of `sdsnew`?

Reply:
thanks @oranagra for your suggestion, updated